src/MonitoringFunctions/DataService/DataServiceFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using MonitoringFunctions.Providers;
+using System;
+
+namespace MonitoringFunctions
+{
+    internal class DataServiceFactory
```

Comment:
Maybe make this class also sealed, since it has no virtual members?

Reply:
Also maybe make this class static? I have only seen the usage:
```c#
new DataServiceFactory().GetDataService()
```

Reply:
Thanks for the suggestions. I am not sure if I feel strongly about either of those. 
Class is already internal and safe to inherit from, even though there may be no reason to do so.
Making it static will save us an allocation, but it will make it harder if we needed instances to configure and pass around in the future.

Reply:
i am a +1 on sealing.  Would like to understand what design do you have in mind to leaving it unsealed?

Reply:
minor issue.  I'm championing for best practices and homogeneity in our team code base.